lib/chef/resource/locale.rb

```diff
@@ -36,56 +36,41 @@ class Locale < Chef::Resource
       action :update do
         description "Update the system's locale."
 
-        if node["init_package"] == "systemd"
-          # on systemd settings LC_ALL is (correctly) reserved only for testing and cannot be set globally
```

Comment:
So there seems to be a comment here describing that the bug report is actually not a bug at all but a deliberate design choice.

See: https://wiki.archlinux.org/index.php/locale#LC_ALL:_troubleshooting

And I tend to agree since setting LC_ALL routinely leads to users not understanding what they're doing and submitting bug reports in all our command line tools when they break UTF-8 encoding (`LC_ALL=C` being the prime offender).

Removing using the public localectl interface for systemd users also seems like a large step backwards, not forwards.